api/v1beta1/metal3machine_types.go

```diff
@@ -87,7 +87,8 @@ func (s *Metal3MachineSpec) IsValid() error {
 	if s.Image.URL == "" {
 		missing = append(missing, "Image.URL")
 	}
-	if s.Image.Checksum == "" {
+	// Checksum is not required for live-iso.
+	if (s.Image.DiskFormat == nil || *s.Image.DiskFormat != "live-iso") && s.Image.Checksum == "" {
```

Comment:
Is not `s.Image.DiskFormat == nil || *s.Image.DiskFormat != "live-iso" <=> *s.Image.DiskFormat != "live-iso" `since `s.Image.DiskFormat == nil `included in `*s.Image.DiskFormat != "live-iso"`

Reply:
All I want to add is to skip Checksum validation when DiskFormat == live-iso. That is verify Checksum only when DiskFormat != live-iso.